File: solve/11723.cs
using System.Text;
var sb = new StringBuilder();
var n = int.Parse(Console.ReadLine());
var s = new HashSet<byte>();
while (n-- > 0)
{
    var spl = Console.ReadLine().Split();
    var cmd = spl[0];
    byte val = 0;
    if (spl.Length == 2)
         val = byte.Parse(spl[1]);

    switch (cmd)
    {
        case "add":
            s.Add(val);
            break;
        case "remove":
            s.Remove(val);
            break;
        case "check":
            sb.AppendLine(s.Contains(val) ? "1" : "0");
            break;
        case "toggle":
            if (!s.Remove(val))
                s.Add(val);
            break;
        case "all":
            s = new HashSet<byte>();
            for (byte i = 1; i <= 20; i++)
                s.Add(i);
            break;
        case "empty":
            s.Clear();
            break;
    }
}
Console.WriteLine(sb.ToString());

File: solve/11723.cpp
// https://www.acmicpc.net/problem/11723

#include <iostream>
#include <cstdio>
#include <algorithm>
#include <string>

using namespace std;

bool s[22];

int main()
{
    ios::sync_with_stdio(false);
    cin.tie(NULL);
    cout.tie(NULL);

    int m;
    cin >> m;
    for (int i = 0; i < m; i++)
    {
        string op;
        cin >> op;

        if (op == "all")
        {
            for (int j = 0; j < 22; j++)
                s[j] = true;
        }
        else if (op == "empty")
        {
            for (int j = 0; j < 22; j++)
                s[j] = false;
        }
        else
        {
            int v;
            cin >> v;

            if (op == "add")
                s[v] = true;
            else if (op == "remove")
                s[v] = false;
            else if (op == "check")
                cout << (s[v] ? "1\n" : "0\n");
            else if (op == "toggle")
                s[v] = !s[v];
        }
    }

    return 0;
}